Library/Dependencies/napi/source/js_native_api_JavaScriptCore.cc

```diff
@@ -1,353 +1,912 @@
+    if (i == 25) method = JSCStaticMethod<25>;
+    if (i == 26) method = JSCStaticMethod<26>;
+    if (i == 27) method = JSCStaticMethod<27>;
+    if (i == 28) method = JSCStaticMethod<28>;
+    if (i == 29) method = JSCStaticMethod<29>;
+    if (i == 30) method = JSCStaticMethod<30>;
+    if (i == 31) method = JSCStaticMethod<31>;
+    if (i == 32) method = JSCStaticMethod<32>;
+    if (i == 33) method = JSCStaticMethod<33>;
+    if (i == 34) method = JSCStaticMethod<34>;
+    if (i == 35) method = JSCStaticMethod<35>;
+    if (i == 36) method = JSCStaticMethod<36>;
+    if (i == 37) method = JSCStaticMethod<37>;
+    if (i == 38) method = JSCStaticMethod<38>;
+    if (i == 39) method = JSCStaticMethod<39>;
+    if (i == 40) method = JSCStaticMethod<40>;
+    if (i == 41) method = JSCStaticMethod<41>;
+    if (i == 42) method = JSCStaticMethod<42>;
+    if (i == 43) method = JSCStaticMethod<43>;
+    if (i == 44) method = JSCStaticMethod<44>;
+    if (i == 45) method = JSCStaticMethod<45>;
+    if (i == 46) method = JSCStaticMethod<46>;
+    if (i == 47) method = JSCStaticMethod<47>;
+    if (i == 48) method = JSCStaticMethod<48>;
+    if (i == 49) method = JSCStaticMethod<49>;
+    if (i == 50) method = JSCStaticMethod<50>;
+    if (i == 51) method = JSCStaticMethod<51>;
+    if (i == 52) method = JSCStaticMethod<52>;
+    if (i == 53) method = JSCStaticMethod<53>;
+    if (i == 54) method = JSCStaticMethod<54>;
+    if (i == 55) method = JSCStaticMethod<55>;
+    if (i == 56) method = JSCStaticMethod<56>;
+    if (i == 57) method = JSCStaticMethod<57>;
+    if (i == 58) method = JSCStaticMethod<58>;
+    if (i == 59) method = JSCStaticMethod<59>;
+    staticFunctions[i] = {methodDescriptors[i].utf8name, method, kJSPropertyAttributeReadOnly | kJSPropertyAttributeDontDelete};
+    table.table[i] = {methodDescriptors[i].method, methodDescriptors[i].data};
+  }
+  staticFunctions[methodCount] = {0, 0, 0};
+
+  classDefinition.staticFunctions = staticFunctions;
+
+
+  JSClassRef classDef = JSClassCreate(&classDefinition);
+  JSObjectRef ctor = JSObjectMakeConstructor(context, classDef, JSCCallAsConstructor);
+  JSObjectSetProperty(context, globalObj, str, ctor, kJSPropertyAttributeNone, nullptr);
+  
+  ClassTable *cbInfo = new ClassTable{cb, data, classDef, env, table};
+  constructorCB[ctor] = cbInfo;
+
+  JSStringRelease(str);
+  *result = reinterpret_cast<napi_value>(ctor);
   return napi_ok;
 }
 
-napi_status napi_get_element(napi_env env,
-                             napi_value object,
-                             uint32_t i,
-                             napi_value* result) {
+napi_status napi_get_array_length(napi_env env,
+                  napi_value v,
+                  uint32_t* result) {
+  assert(JSValueIsArray(env->m_globalContext, reinterpret_cast<JSObjectRef>(v)));
+  
+  bool hasLength = JSObjectHasProperty(env->m_globalContext, reinterpret_cast<JSObjectRef>(v), JSStringCreateWithUTF8CString("length"));
```

Comment:
this should be inlined with the assert if the intention is to not check it in release so that it won't execute JSObjectHasProperty in release

Reply:
Fixed